### DISCUSSION TITLE
Adding non-UI fields to "classes"

### DISCUSSION THREAD

**Discussion by martinellison** [fltk-rs/fl2rust#34] [Ideas]
This is more of a discussion point rather than an issue, but I can't find a discussion forum for this project...

Is there any way to add lines of code to a "class" i.e. a struct? I wanted to add some field declarations but the "declaration" item puts the code somewhere else.

**Comment by MoAlyousef** [comment]
Currently fl2rust just deals with global declarations, that's why you find your declarations are moved to the module's namespace. Your suggestion is useful, however it would be hard to implement correctly. Any declaration in a struct would need to be properly initialized in a constructor function. I'll try to have a go at it at some point.

**Comment by martinellison** [comment]
Thanks for replying. At the moment, I am still learning FLTK and FLUID; I think that now I can code up what I need (I have now realised that I can write an extra code file that does anything that FLUID/fl2rust does not generate).

Actually, the most useful things at the moment would be

* a discussion area so the we don't need to use the issues list to ask newbie questions, and
* some documentation for fl2rust so we know what it can and cannot do and how to use FLUID to make fl2rust geneerate what it can generate.
